frontend/src/components/DataTableComponent.tsx

```diff
@@ -1,10 +1,11 @@
 import React from 'react';
 import DataTable, { SortOrder, TableColumn } from 'react-data-table-component';
+import { classNames } from '../common/utils/utils';
 
 interface DataTableProps<T> {
-  className: string;
+  className?: string;
   columns: TableColumn<T>[];
-  data: T[];
+  data?: T[];
   rowsPerPageText?: string;
   rangeSeparatorText?: string;
   EmptyContent?: JSX.Element;
```

Comment:
I think that we will make this consistent throughout the app so we should have the LoadingContent, EmptyContent and SortIcon set inside the component as it would be the same for all the instances where we will use this table.